### PR TITLE
Touch up Server.cs

### PR DESCRIPTION
**Pull request by penev92**
The first commit contains what I believe is important logging on error (it is very annoying when you try to start a skirmish game and you keep getting kicked from the server as soon as you try to connect, with _no_ message and _no_ logs).
The second commit contains exactly what its name says. If anyone has any requests for something else in that file that they would like to see beautified, now is the time.

### CODE REVIEW

**Comment by Mailaender** [comment]
:+1:

File: OpenRA.Game/Server/Server.cs [hunk 1]
```diff
@@ -362,34 +374,36 @@ void ValidateClient(Connection newConn, string data)
...
-					"and may desynchronize the game state if they have incompatible rules.");
+						"and may desynchronize the game state if they have incompatible rules.");
 
 				SetOrderLag();
 			}
-			catch (Exception) { DropClient(newConn); }
+			catch (Exception ex)
+			{
+				Log.Write("server", "Dropping connection {0} because an error occured:{1}{2}",
+					newConn.Socket.RemoteEndPoint, Environment.NewLine, ex);
```
Comment (obrakmann) [inline OpenRA.Game/Server/Server.cs]: Log messages should not contain new lines, IMHO.

**Comment by obrakmann** [comment]
Looks ok otherwise, :+1: . For the log, I suggest two messages:
```
Log.Write("server", "Dropping connection {0} because an error occurred. See exception.log for the details.", newConn.Socket.RemoteEndPoint)
Log.Write("exception", ex)
```
(not sure if the syntax works on that second line)

**Comment by penev92** [comment]
Changed.

**Comment by penev92** [comment]
Changed again.

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/6d6bb2076c24ecbdf0ab497f9845d7033ca1d1a2%5E...6d6bb2076c24ecbdf0ab497f9845d7033ca1d1a2)

File: OpenRA.Game/Server/Server.cs [hunk 2]
```diff
@@ -673,13 +674,10 @@ public void StartGame()
 				DropClient(c);
 
 			// Drop any players who are not ready
-			foreach (var c in Conns.ToArray())
+			foreach (var c in Conns.Where(c => GetClient(c).IsInvalid))
```
Comment (RoosterDragon) [inline OpenRA.Game/Server/Server.cs]: This will break when ```DropClient``` removes somebody from the ```Conns``` list!

**Pull request merged**